### DISCUSSION TITLE
How to add managed metadata column value in details layout of pnp search results webpart

### DISCUSSION THREAD

**Discussion by supreet-bajaj** [microsoft-search/pnp-modern-search#3289] [Q&A]
I have been trying to add managed metadata column value in details layout of pnp search results webpart but its not showing term value properly. Even its not showing term values for multiple select allowed in Managed metadata column.
![IMG_6229](https://github.com/microsoft-search/pnp-modern-search/assets/25117777/c0ec337f-2a60-462a-bb9b-2517c3911abb)

Can someone please provide me solution how I can show multiple and single value of managed metadata column. 

is there handlebars expression which I need to use?

**Comment by kasperbolarsen** [comment]
Hi, just to make sure: Are you requesting a sample that shows how to render the label values using handlebars?

**Comment by supreet-bajaj** [comment]
Hi @kasperbolarsen , 

thanks for your reply. I am looking for solution how I can show label values from managed metadata (single value and multiple values) column which is setup in SharePoint Search schema using its crawled property.

**Comment by kasperbolarsen** [comment]
Hi,
Should be possible without any handlebars at all. I have mapped the crawled property to a refinable string, and is using that refinable string in the Layout :
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/3e5f240c-6a61-4224-85ce-4da7f931bef1)


![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/cb8e8ab1-48a3-4a4c-9844-67a06673524c)


![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/9d087ee8-1e28-481c-aad3-58afdeca6b22)

**Comment by supreet-bajaj** [comment]
Thanks @kasperbolarsen for letting me know.. Is there a way to replace semi-colon with a comma between two managed metadata column label values?

**Comment by kasperbolarsen** [comment]
Hi @supreet-bajaj , if you want to replace the ; you will have to use Handlebars
I am using this expression in a custom Layout, see below
{{#each (split (slot item @root.slots.TaxMulti) ";") as |tag| }}
                                                        {{last (split tag "|")}}

And the TaxMulti field is rendered in the "Retention" block:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/50cdfd55-269c-445b-9ca2-cf68600537aa)

**Comment by supreet-bajaj** [comment]
Hi @kasperbolarsen ,

I have tried the handlebars expression which you mentioned, but its showing below error.
Here I have used [RefinableString02] in handlebars expression. Please let me know if I did something wrong.
![HandlebarError](https://github.com/microsoft-search/pnp-modern-search/assets/25117777/e5dd74f3-435c-400b-8c15-c37b09dc1108)
![Results Webpart](https://github.com/microsoft-search/pnp-modern-search/assets/25117777/e5a7227c-e7b5-4585-bb98-bdfe3dac5306)

Thanks.

**Comment by kasperbolarsen** [comment]
Please note that I am not refering the field by its Managed Property name but by its Title in the Layout slots ( top of "page" 1 of the Results web part:  
( see https://microsoft-search.github.io/pnp-modern-search/usage/search-results/slots/ for reference)

![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/624287f7-a149-46b3-b14c-86df5bab2a88)
